src/uu/cp/src/copydir.rs

```diff
@@ -292,6 +299,7 @@ fn copy_direntry(
                 Err(e) => return Err(e),
             }
         }
+        seen_sources.insert(source_relative.file_name().unwrap().into(), local_to_target);
```

Comment:
This can't be right can it? What happens with multiple files with the same file name but at different locations?